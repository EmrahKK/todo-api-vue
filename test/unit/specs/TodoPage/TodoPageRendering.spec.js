import Vue from 'vue'
import Vuex from 'vuex'
import { mount } from 'vue-test-utils'
import TodoPage from '@/components/Todo/TodoPage'

Vue.use(Vuex)

describe('Todo page component rendering:', () => {
  let getters
  let store
  let actions

  beforeEach(() => {
    getters = {
      todoList: state => state.todo,
      newTodoName: state => state.newTodoName,
      isLoading: state => state.isLoading
    }

    actions = {
      getTodos: jest.fn()
    }
  })

  it('Renders todo page with no results', () => {
    store = new Vuex.Store({
      getters,
      actions
    })

    const wrapper = mount(TodoPage, { store })
    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })

  it('Should have some list items', () => {
    store = new Vuex.Store({
      getters,
      actions,
      state: {
        'todo': [{
          'id': 5,
          'name': 'Hacer la colada oscura',
          'isComplete': false
        }, {
          'id': 6,
          'name': 'Comprar matadinosaurios',
          'isComplete': true
        }, {
          'id': 19,
          'name': 'Barrer el suelanis',
          'isComplete': false
        }, {
          'id': 32,
          'name': 'foobaris',
          'isComplete': true
        }, {
          'id': 34,
          'name': 'Urujai',
          'isComplete': false
        }]
      }
    })

    const wrapper = mount(TodoPage, { store })
    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })

  it('Should match input value with state value', () => {
    store = new Vuex.Store({
      getters,
      actions,
      state: {
        'newTodoName': 'some todo name'
      }
    })

    const wrapper = mount(TodoPage, { store })
    const input = wrapper.find('input')

    expect(input.element.value).toEqual('some todo name')
  })

  it('Should disable input and buttons when is on loading state', () => {
    store = new Vuex.Store({
      getters,
      actions,
      state: {
        'todo': [{
          'id': 5,
          'name': 'Hacer la colada oscura',
          'isComplete': false
        }],
        'isLoading': true
      }
    })

    const wrapper = mount(TodoPage, { store })
    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })
})
