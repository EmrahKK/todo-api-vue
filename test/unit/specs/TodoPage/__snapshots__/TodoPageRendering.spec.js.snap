// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo page component rendering: Renders todo page with no results 1`] = `
<div>
    <div class="main">
        <h1>Todo list</h1>
        <div class="content">
            <form>
                <div class="form-group flexible">
                    <input class="form-control">
                    <button class="btn btn-primary">Set new todo</button>
                </div>
            </form>
            <ul></ul>
        </div>
    </div>
</div>
`;

exports[`Todo page component rendering: Should disable input and buttons when is on loading state 1`] = `
<div>
    <div class="main">
        <h1>Todo list</h1>
        <div class="content">
            <form>
                <div class="form-group flexible">
                    <input disabled="disabled" class="form-control">
                    <button disabled="disabled" class="btn btn-primary">Set new todo</button>
                </div>
            </form>
            <ul>
                <li>
                    <label>
                        <input type="checkbox"> <span class="">
        Hacer la colada oscura
      </span></label>
                    <button disabled="disabled" class="btn btn-xs btn-danger">X</button>
                </li>
            </ul>
        </div>
    </div>
</div>
`;

exports[`Todo page component rendering: Should have some list items 1`] = `
<div>
    <div class="main">
        <h1>Todo list</h1>
        <div class="content">
            <form>
                <div class="form-group flexible">
                    <input class="form-control">
                    <button class="btn btn-primary">Set new todo</button>
                </div>
            </form>
            <ul>
                <li>
                    <label>
                        <input type="checkbox"> <span class="">
        Hacer la colada oscura
      </span></label>
                    <button class="btn btn-xs btn-danger">X</button>
                </li>
                <li>
                    <label>
                        <input type="checkbox"> <span class="tacher">
        Comprar matadinosaurios
      </span></label>
                    <button class="btn btn-xs btn-danger">X</button>
                </li>
                <li>
                    <label>
                        <input type="checkbox"> <span class="">
        Barrer el suelanis
      </span></label>
                    <button class="btn btn-xs btn-danger">X</button>
                </li>
                <li>
                    <label>
                        <input type="checkbox"> <span class="tacher">
        foobaris
      </span></label>
                    <button class="btn btn-xs btn-danger">X</button>
                </li>
                <li>
                    <label>
                        <input type="checkbox"> <span class="">
        Urujai
      </span></label>
                    <button class="btn btn-xs btn-danger">X</button>
                </li>
            </ul>
        </div>
    </div>
</div>
`;
